### PR TITLE
Fix: add json data into registry page

### LINKED ISSUE #12213: [UI] Show JSON data into registry page
### Current Behavior
Currently, we are showing only limited data when selecting any modal or component on the registry page. 

### Desired Behavior
Show All the JSON data for the selected Item.

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by amitamrutiya**
**Notes for Reviewers**

This PR fixes #12213 

image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/MeshModelRegistry/MeshModelDetails.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ import {
 import _ from 'lodash';
 import { JustifyAndAlignCenter } from './MeshModel.style';
 import { withSuppressedErrorBoundary } from '../General/ErrorBoundary';
+import ReactJson from 'react-json-view';
```
Comment (sudhanshutech) [inline ui/components/MeshModelRegistry/MeshModelDetails.js:30]: why we using a new package. Can we use the existing codemirror package to use here?

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: ui/components/MeshModelRegistry/MeshModelDetails.js (hunk 1 above)
Comment (amitamrutiya) [reply ui/components/MeshModelRegistry/MeshModelDetails.js:30]: @aabidsofi19, recommend using this library.

Comment (aabidsofi19) [reply ui/components/MeshModelRegistry/MeshModelDetails.js:30]: yeah , we use it in extensions already

**Comment by leecalcote** [comment]
@amitamrutiya there needs to be a way to show/hide these advanced details

**Comment by amitamrutiya** [comment]
Okay I will add an accordion for that if it makes sense.

**Comment by amitamrutiya** [comment]
@leecalcote I hope I delivered right thing that you want.
[Screencast from 2024-10-05 10-59-51.webm](https://github.com/user-attachments/assets/1b2cf026-d2df-4913-b9c2-a780bc51a26c)

File: ui/components/MeshModelRegistry/MeshModelDetails.js [hunk 2]
```diff
@@ -27,7 +27,12 @@ import {
 import _ from 'lodash';
 import { JustifyAndAlignCenter } from './MeshModel.style';
 import { withSuppressedErrorBoundary } from '../General/ErrorBoundary';
-
+import ReactJson from 'react-json-view';
+import { reactJsonTheme } from './helper';
+import Accordion from '@material-ui/core/Accordion';
+import AccordionDetails from '@material-ui/core/AccordionDetails';
+import AccordionSummary from '@material-ui/core/AccordionSummary';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
```
Comment (aabidsofi19) [inline ui/components/MeshModelRegistry/MeshModelDetails.js:35]: why are we not using sistent .

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: ui/components/MeshModelRegistry/MeshModelDetails.js (hunk 2 above)
Comment (aabidsofi19) [reply ui/components/MeshModelRegistry/MeshModelDetails.js:35]: please use sistent . we constantly regress back and end up churning too much.  . 
All new and refactored components should strictly use sistent .

Comment (amitamrutiya) [reply ui/components/MeshModelRegistry/MeshModelDetails.js:35]: I will keep that in my mind.

**Comment by leecalcote** [comment]
> @leecalcote I hope I delivered right thing that you want. [Screencast from 2024-10-05 10-59-51.webm](https://github.com/user-attachments/assets/1b2cf026-d2df-4913-b9c2-a780bc51a26c)

Thanks for this recording, @amitamrutiya. This looks pretty good to me. :thumbs-up:

**Pull request merged**